sys/net/gnrc/network_layer/sixlowpan/frag/gnrc_sixlowpan_frag.c

```diff
@@ -263,46 +263,36 @@ void gnrc_sixlowpan_frag_send(gnrc_pktsnip_t *pkt, void *ctx, unsigned page)
     if (fragment_msg->offset == 0) {
         /* increment tag for successive, fragmented datagrams */
         _tag++;
-        if ((res = _send_1st_fragment(iface, fragment_msg->pkt, payload_len, fragment_msg->datagram_size)) == 0) {
+        if ((res = _send_1st_fragment(iface, fragment_msg->pkt, payload_len,
+                                      fragment_msg->datagram_size)) == 0) {
             /* error sending first fragment */
             DEBUG("6lo frag: error sending 1st fragment\n");
-            gnrc_pktbuf_release(fragment_msg->pkt);
-            fragment_msg->pkt = NULL;
-            return;
+            goto error;
```

Comment:
When I learned programming C over 30 years ago, it was an absolute nogo to use `goto` statements. Our lecturer said, `goto` breaks the execution structure and if you need the `goto` statement, there is something wrong with your structure. He said also `goto` is very bad and something like `BASIC`. It must not be used.

Has this theory changed?

Even it seems sometimes easier to have a goto, especially for exits on error, I never used it all the years. I know, it avoids code duplication.

Reply:
It is ok to me to leave the `goto` statement here. I think it is very comfortable to have a single exit point to avoid code duplication. The alternative would be to have very deep `if () ... else if ... ` structures which doesn't improve the readability of source code.

Reply:
Removed the gotos.

Reply:
Ups. Sorry, it should be just a question no change request.

Reply:
Sorry if it sounded like a change request.

Reply:
Welp, now I do the same, just without goto ;-)